### ISSUE TITLE
keyerror creating a new env from a tarball

### ISSUE THREAD

**Issue by ReimarBauer** [conda/conda#2970]
Hi all

I ran into this problem

conda create -n mssdev /home/user/miniconda2/conda-bld/linux-64/mss-1.2.0-np111py27_0.tar.bz2

```
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/home/user/miniconda2/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/home/user/miniconda2/lib/python2.7/site-packages/conda/cli/main.py", line 120, in main
    exit_code = args_func(args, p)
  File "/home/user/miniconda2/lib/python2.7/site-packages/conda/cli/main.py", line 130, in args_func
    exit_code = args.func(args, p)
  File "/home/user/miniconda2/lib/python2.7/site-packages/conda/cli/main_create.py", line 57, in execute
    install(args, parser, 'create')
  File "/home/user/miniconda2/lib/python2.7/site-packages/conda/cli/install.py", line 203, in install
    explicit(args.packages, prefix, verbose=not args.quiet)
  File "/home/user/miniconda2/lib/python2.7/site-packages/conda/misc.py", line 126, in explicit
    index.update(get_index(**index_args))
  File "/home/user/miniconda2/lib/python2.7/site-packages/conda/api.py", line 22, in get_index
    channel_urls = normalize_urls(channel_urls, platform, offline)
  File "/home/user/miniconda2/lib/python2.7/site-packages/conda/config.py", line 250, in normalize_urls
    url = urls[0]
KeyError: 0

```

```
Current conda install:

             platform : linux-64
        conda version : 4.1.6
    conda-env version : 2.5.1
  conda-build version : 1.21.3
       python version : 2.7.12.final.0
     requests version : 2.9.1
     root environment : /home/user/miniconda2  (writable)
  default environment : /home/user/miniconda2
     envs directories : /home/user/miniconda2/envs
        package cache : /home/user/miniconda2/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
         offline mode : False
    is foreign system : False
```

If I first create a env by 

`conda create -n mssdev python=2`
and afterwards build from that activated env I get the package installed

**Comment by mcg1969** [comment]
Thanks for the report, and I'm glad you have a workaround.

**Comment by jjhelmus** [comment]
Related to #2974

**Comment by kalefranz** [comment]
fixed with #2976
